### ISSUE TITLE
[Issue]: Australian Classification Ratings not correct

### ISSUE THREAD

**Issue by starkebn** [jellyfin/jellyfin#11650]
### Please describe your bug

I believe this file needs editing: 

https://github.com/jellyfin/jellyfin/blob/fd250e4fe1c1a35e8e40c008a7d25675729e3dac/Emby.Server.Implementations/Localization/Ratings/au.csv

The Australian Classification has 6 ratings:

https://www.classification.gov.au/classification-ratings/what-do-ratings-mean

- G: no age restriction
- PG: PG-rated content is not recommended for viewing by people under the age of 15 without guidance from parents, teachers or guardians.
- M: M-rated films and computer games are not recommended for children under the age of 15. However, children under the age of 15 may legally access this material as the classification rating is an advisory category.
- MA 15+: legally restricted to people over the age of 15
- R 18+: legally restricted to people over the age of 18
- X 18+: legally restricted to people over the age of 18

in Jellyfin 1.0.9.0+ the list currently looks like the attached screenshot.

PG and M should be before MA 15+, probably at age 14. I'm not sure if the file needs all the other age ratings that do not appear in the Australian classification system, I don't think they appeared in 1.0.8.0 and below.

### Reproduction Steps

Jellfin > Dashboard > Users > Parental Control > Maximum Allowed Parental Rating dropdown

### Jellyfin Version

10.9.0

### Environment

```markdown
any
```

### Jellyfin logs

```shell
none needed
```

### Please attach any screenshots here

![image](https://github.com/jellyfin/jellyfin/assets/36748864/6e7755b2-7fac-49e2-b8d6-932d330df4ac)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Shadowghost** [comment]
Classifications were sourced from multiple sources: IMDB, Wikipedia and the one you linked.

From what I can see the only thing requiring changes is PG being 16 instead of 15

**Comment by starkebn** [comment]
at minimum it should be:

G; 0
PG: 14
M: 14
MA 15+: 15
R 18+: 18
X 18+: 18

I'm not sure anything else is necessary, but PG and M should be before MA 15+ at the very least

**Comment by Shadowghost** [comment]
Fine for me. But I think X18+ should be kept at 1000 (this level is used for X-rated content/pornography)

**Comment by danmolloy** [comment]
Hi there,
Just saw this conversation and thought I might add my two cents (I'm Australian).
M-rated films are advised for audiences 15+, whereas audiences legally must be 15+ for MA. PG (parental guidance) is fine for children, but they may need an adult to explain some themes.

You might find [Community Cinema](https://communitycinemas.com.au/classification-ratings/#:~:text=M%20is%20an%20advisory%20classification,for%20people%20aged%20under%2015.)'s page helpful.

**Comment by Shadowghost** [comment]
When implementing this I opted for being conservative and use the advised age as the limit - because there is no way for us to differentiate if a parent is watching it with their kid.

**Comment by Fenruz** [comment]
I found this GitHub Issue after upgrading to 10.9.x and finding most of my kids library was missing. I am also in Australia and the current ratings list (the same as the one in the screenshot above) is not reflective of our rating system. As it is now the function in Jellyfin is broken. If I set it to <14 it barely shows 10% of their library. Set to PG/16+ it shows inappropriate content. This worked perfectly pre-upgrade. Please fix if you can, I'm happy to post screenshots if they're allowed from the GUI demonstrating the issue.

**Comment by starkebn** [comment]
yes, something still needs to be fixed here

**Comment by starkebn** [comment]
I'm not sure how it works, but it needs to be changed so that if you restrict an account to PG it can't see M or above, and if you restrict it to M the account can't see MA 15+ etc.

The list of restriction priorities is in the initial post

**Comment by Shadowghost** [comment]
But that is the main problem here:
PG: 15+ since we can't validate if a parent is watching?
M: 15+ but it's advisory
MA 15+ legally enforced 15+

So I guess we'd need to lower PG to 14 to not make it include M or MA content...

**Comment by Fenruz** [comment]
I'm happy to try it if you let me know how :-) I'd be very grateful! As it stands now it's basically broken and can't be used by the family :-( I set "14" but I guess because this doesn't match Australian ratings it shows nothing, while PG/16+ shows Peppa Pig alongside Joker :-( All the best.

**Comment by starkebn** [comment]
it's important that it's useful in the Parental Control section of Jellyfin, I don't care about anything else

the priority list is G < PG < M < MA 15+ < R18+ and maybe X 18+

just like how it used to be and was useful

right now it's a mess and it sucks

**Comment by oddstr13** [comment]
If you are really particular about which shows your kids can access, then use the tagging feature.
There is both a whitelist (only items containing tags) and a blacklist (don't allow items with tags).

For the age rating, we *have* to assume the kid is watching the content alone.

Also be warned that the parental controls only recently have started becoming somewhat useful, and that there are likely still edge cases that doesn't work properly. (Auto-generated library thumbnails can be from any content in the library for example).

**Comment by starkebn** [comment]
Why such big pushback here? It was working great in 1.0.8 and now it's almost non functional. The Australian classification system is very simple and the current implementation is a jumble of ratings from many different areas that seem unnecessary and it's not useful. It was changed for not yet explained reasons and to me it seems like it's worse.

**Comment by Fenruz** [comment]
Dear @starkebn, I think we need to be nice to the Dev's, this is an open-source project and I believe they donate their time, so from all of us (especially down under) thank you very much!

Dear @oddstr13, I do appreciate @starkebn position however, this wasn't starting to become useful as you describe, at least not to us, it was wonderful & perfectly functional prior to the updates. There are work-arounds as you describe, another example is I could separate their content out into their own folders, but it's these working features of the software (even free software like this) that make it such a pleasure to use! 

Thank you sincerely for committing the change, I'll track it and install the update ASAP when it gets rolled in. Not all hero's wear capes, but you do mate!

**Comment by oddstr13** [comment]
Explaination in the PR implementing the overhaul #8526, which fixed a bunch of issues with parental controls (see list in PR description).

The fact that it worked for you in 10.8 seems to have been pure luck, as it often wasn't working at all before it got reworked.

**Comment by Shadowghost** [comment]
To add some context to what and why things were changed:
Before 10.9 rating strings (PG,MA, etc) were assigned a number and those were compared to each other for access control. This worked somewhat fine if your rating strings came from only one system (emphasis on somewhat) but totally broke once you had rating strings from multiple systems in your library because the number usage was not consistent between the rating system definitions in Jellyfin. #8526 redefined all `rating string -> rating level` mappings to be consistent and implemented proper enforcement of the restrictions. Generally speaking, before that PR any kind of working parental control based on ratings was at best somewhat working but incomplete and at worst not working as intended at all.
Technically speaking PG, MA and M are all 15+ because we simply do not know that whoever is watching has a parent next to them. The proper fix for this is to introduce a kind of sublevel which allows differentiation of such cases (e.g. major level is 15, sublevel for PG is 0, MA is 1 and M is 2). Since this requires API changes, this will not happen in 10.9 though.
#12043 at least correctly sets the level to 15 instead of 16 but there really is not more we can do for the 10.9 release cycle.

**Comment by starkebn** [comment]
> "Technically speaking PG, MA and M are all 15+ because we simply do not know that whoever is watching has a parent next to them."

I personally think this is not the most useful way of looking at the classifications. It doesn't feel very functional. In the screenshotted section of the dashboard, the intention is that the admin of the server can restrict certain users to certain classification categories. If you insist on putting them all on one line because technically they're based on age, the function becomes unusable. I don't see how focusing on the age of the viewer is helpful.

I can guess from the list you're trying to capture all the random ratings that can be pulled in automatically from IMDB / TVDB / TVMaze etc.

The classification categories are intended to be separated by content, I think it would be better to have the back end rating levels have a separate extra 'priority' field decoupled from age if the current implementation is the result.

just spitballing:

```
000 | 00 | Exempt / G
010 | 07 | 7+
020 | 10 | 10
030 | 13 | 13
040 | 14 | 14
050 | 15 | PG
060 | 15 | M
070 | 15 | MA / MA15+ / MA 15+
080 | 16 | 16+
090 | 18 | Unrated
100 | 18 | R / R18+ / R 18+ / 18+
800 | 18 | X / X18+ / X 18+
900 | 99 | RC
```

**Comment by Shadowghost** [comment]
The reason why the age is the deciding factor is that this is the only thing we can use to match between different rating systems.

**Comment by starkebn** [comment]
Unfortunately matching only by age takes away the intent of content based classification if you insist on putting categories like PG, M and MA 15+ all on the same line.

Going by what you have said this isn't going to be changed in 10.9 so maybe putting the Australian ratings on the following ages would be a messy solution

0: G
14: PG:
15: M
16: MA 15+
18: R 18+:
19: X 18+

**Comment by oddstr13** [comment]
How would you as a parent feel about your 14 year old suddenly gaining access to content rated for 15 year olds?
Because that is potentially the result of changing that value to 14.

The following documentation should get you up and running compiling your own custom version of Jellyfin.
https://github.com/jellyfin/jellyfin?tab=readme-ov-file#server-development
https://github.com/jellyfin/jellyfin-packaging?tab=readme-ov-file#quickstart

I couldn't find the CSV files in the clear on my system, so presumably they're embedded in a dll somewhere.

**Comment by starkebn** [comment]
I can't speak for everyone, but that's a pretty mild problem. The "legal age" part of the classification is the lowest interest to me.

I'll state my point one last time and leave it.

If I tell Jellyfin to restrict an account to PG content, I want PG and G content available, and I don't want any M, MA 15+, R 18+ etc content displayed. If the function works differently then for me it might as well not be there.

**Comment by Fenruz** [comment]
> How would you as a parent feel about your 14 year old suddenly gaining access to content rated for 15 year olds? Because that is potentially the result of changing that value to 14.

Dear oddstr13, I think you're a dead-set legend for working on this project, so _please_ believe me when I say my response here is meant with the kindest of intent and to add to the discussion.

To answer your question as a parent of two in Australia using our rating system - I wouldn't care in the slightest. You are describing a black-and-white binary situation where a setting of 14 vs 15 would deliver staggeringly different content, but the classification system isn't anything like that. It's a guide, a rule of thumb, a suggestion to the audience & parents on the appropriateness of the content. 

Not to mention as individuals, especially younger audiences, can often have a maturity age several years ahead or behind their actual age. The idea that a filter at 14 vs 15 would be such a huge gap doesn't fit with the rating system down-under or the reality of its use. 

I'll give you a real world example! I've found with my kids who are 9 that setting to PG (or age 13) is about right, it let's them see shows like Shrek, Lego Ninja's and the Super Mario Bro's movie while keeping the adult stuff hidden. They're only 9 years old, if I set their age to 9 the idea they couldn't watch Shrek is nuts. The same age filter on things like YouTube is also about-right (of course, parenting comes 1st! I'm always keeping an eye on them. haha)

Please let me know if this rambling helped! I'm sincerely not trying to be argumentative, I hoped the above might be seen as objective. Basically what starkebn concluded with in his post above is right, I just thought I'd add my 2c.

Stay safe & happy, peace.

**Comment by drfarmerdave** [comment]
I noticed this today also. It does seem logical to me that even if its assumed a child is watching alone, if we set the Parental control to PG the library should not show them any MA 15+ content. As stated above
G < PG < M < MA 15+

**Comment by Redjellyfin** [comment]
While I appreciate the work put into this program, as an Australian, the ratings change has made it more dangerous for kids, not better. I had to all of a sudden go through and change all my rating from default to what I felt was appropriate. System was fine before and now is a far worse problem. PG should never be above M or MA15. Please at the very least, move the PG to before the M.

**Comment by oddstr13** [comment]
Larger changes to the rating system is out of scope for 10.9 patches, and will have to wait for 10.10 (or later).

**Comment by Fenruz** [comment]
I just installed v10.9.7, and it's worse... 

Why is PG/M/MA/MA1+/MA 15+ all on the same line?

**Comment by starkebn** [comment]
Because this is now in 10.10 to do list I have brainstormed what I think might be a potential fix for this.

Instead of the ratings being in a strict priority order would it be possible to implement it as a list of checkboxes of allowed / denied access? So each potential rating can be controlled for the account at the admin's preference, and the age recommendation would have nothing to do with it.

**Comment by Shadowghost** [comment]
As mentioned before, the plan is to add sublevels (meaning rating is not represented by an int but by a double). This allows sorting of rating within the same age restriction (e.g. PG would be 15.1, MA 15.2 and M 15.3, etc)

**Comment by starkebn** [comment]
I guess that's fine and should fix it. But why so committed to it being dictated by age? It's basically just a tag set by the regional government. And something that is classified 14 in one region might be classified 18+ in another.

**Comment by Shadowghost** [comment]
Because the only info we get from metadata providers are rating strings. And if you have different providers or even different languages in different libraries, we need to somehow still enforce the proper rating limits on those. Therefore we map all ratings to a "level" which currently just correlates with the age.

**Comment by starkebn** [comment]
Like I said, the system you propose should be fine, but for the sake of discussion: if you treat them as tags admins are able to make their own choices and the coders don't have to worry about "getting it right"

**Comment by starkebn** [comment]
Just so you know, a lot of series import ratings that aren't on your current list. Things I've seen recently: TV-MA, TV-PG, TP, 6 and more.

**Comment by Shadowghost** [comment]
That's exactly why we have the mapping to age. Those rating strings are US strings. If we don't find the strings in the library language rating mappings, we run it through all available rating mappings - thus picking up the US one and mapping it appropriately to an age - allowing us to handle it properly.

**Comment by Fenruz** [comment]
I am looking forward to when you can do it properly, thank you @Shadowghost you're a legend. We all appreciate your work. 

If it's back to how it was pre-update that'd be super, because right now my kids can see Madagascar alongside Kill Bill: Vol. 1, there's something very very broke right now hahaha :-)

**Comment by conradp24** [comment]
I agree with this topic. My daughter is 8. I'm trying to change everything to a classification of 10 years old that she is enabled to watch. Most of the films are classified PG. There is no classification in Jellyfin for AU-PG, which a Google Search suggests that its suitable for an 8 year old. Trying to do this manually is incredibly time consuming. I'm not complaining, its more an observation. I think, if classifications cannot be changed, it would be great to see a mass meta editor where we can tick all the media and have a drop down box to select the classification age.

**Comment by Fenruz** [comment]
Well @conradp24 I have great news!

I've been checking for the release nearly daily since @starkebn said:

> Because this is now in 10.10 to do list I have brainstormed what I think might be a potential fix for this.

We're currently on v10.9!!!

It was such a perfect system prior to blowing up the whole rating system, I hadn't so much as looked back at Plex until it was lit on fire. I'm very much looking forward to my kids being able to use it again.

**Comment by Rygle** [comment]
I am in Australia and had this same problem.

I did a workaround as I have described on the forum and in another github bug;
https://github.com/jellyfin/jellyfin/issues/11767#issuecomment-2335006162

https://forum.jellyfin.org/t-parental-rating-certification-list-order?pid=35645#pid35645

I think the **real solution** is to do all parental rating as per the country's rating system for anything where there is any ambiguity - if the country has a rating of AU-PG, then that should be a completely separate rating from PG-13, 13, M, AU-M, AU-MA15+, 15, 14. If the person has chosen a parental rating of AU-PG in Jellyfin, then anything that is in a PG related classification should not show up unless the administrator has properly identified it as AU-PG.

So, if I set my child's classification/parental rating to AU-PG, the following will apply;

U - will show for AU-PG
G - will show for AU-PG
AU-PG will show for AU-PG
PG, PG-13, 13, 14, M, MA, AU-M, AU-MA15 and anything else that is not explicity U, G or AU-PG will not show up until I get the correct rating in the metadata

**Comment by Rygle** [comment]
This whole thing seems have gone wrong in an effort to make things more granular.

Perhaps a better approach would be to make things more granular **in the settings** rather than in the **interpretation of the settings**. So all profiles have a check list of all possible ratings, and the parent/administrator has to explicitly tick every rating they want to be visible. Or to speed this up, it could have umbrella categories with sub-categories that get selected if the umbrella category is selected, but *only* if the umbrella category is selected. Alternately, there could be a tick box for "more granular ratings selections" that does something like the following (if unticked it can remain as it was)...

So, in the Jellyfin settings, I will see something like;

NR [ ]
-- NR [ ]
-- AU-NC [ ]
U [ ]
-- U [ ]
-- AU-U [ ]
etc for other countries

G [ ]
-- MPAA-G  [ ]
-- AU-G [ ]
-- 7 [ ]
etc for other countries

PG [ ]
-- 13
-- PG-13 [ ]
-- AU-PG [ ]
-- 14  [ ]
etc for other countries

M [ ]
-- MPAA-M [ ]
-- 15 [ ]
-- AU-M [ ]
-- AU-MA15 [ ]
etc for other countries

etc, etc.

That way a parent/administrator can tick either the larger

**Comment by starkebn** [comment]
> This whole thing seems have gone wrong in an effort to make things more granular.
> 
> Perhaps a better approach would be to make things more granular **in the options available in the settings** rather than **in the interpretation of the settings** that the user has no control over (apart from re-compiling Jellyfin). So all profiles have a check list of all possible ratings, and the parent/administrator has to explicitly tick every rating they want to be visible. Or to speed this up, it could have umbrella categories with sub-categories that get selected if the umbrella category is selected, but _only_ if the umbrella category is selected. Alternately, there could be a tick box for "more granular ratings selections" that does something like the following (if un-ticked it can remain as it was)...

yep, this is what I also suggested so I'll say I support this

**Comment by Shadowghost** [comment]
Since people keep commenting on any kind of issue or PR concerning this, I'll summarize the current state:
* Yes, we are aware of the issue
* No, it's not as simple as setting a different level for PG because that breaks interoperability between the rating systems
* Making ratings user configurable is not an option due to interoperability

The proposed solution I'm currently working on:
[X] Extend the rating definition with a Sub score
[ ] Only consider subscore if the localized rating system of a user uses subscribed (this is the complex part and required for the interoperability)
[ ] Modify the database and sorting queries to support this (time consuming work)
[x] Rewrite all existing rating system definitions to fit the above

Due to time constraints on my side 10.10 will likely only contain API changes without them being backed by the new data, so clients can migrate.

One last note on what "interoperability" means:
If you metadata provider does not provide a rating in your configured language (this usually means it falls back to either US or original release country rating), we need to parse that rating. Since it's not included on your native rating system, we're doing a lookup on the other rating systems. If now the levels between those systems are not uniform, the resulting level will be useless for you.
This happens way more often than you expect for non-English metadata.

**Comment by Fenruz** [comment]
@Shadowghost you're an absolute legend for donating your time to this project, sincerely thank you and I'm sure we all look forward to the fix!

I wanted to comment just from an end-user perspective that this isn't some minor tweak we'd like to see, it's a huge component of why I use Jellyfin / Plex, in fact I've stopped using it now because it's critical to my use-case. 

Everyone has their own "must-have" I suppose, but for me it's this. For others it would be good subtitles. Frankly I could take or leave the nice thumbnails or resume where I stopped feature. I was using DLNA before this on my 15 year old TV and it was totally fine, this just gave me a chance to share my collection with my kids easily. I've turned Jellyfin off so they don't access things they shouldn't, and am using my old DLNA setup again :-)

I would have just rolled back to the previous version but it said the Android App would stop supporting it, so that put a stop to that. 

I'll keep checking in here, I'm not in a rush, it'll be great to use Jellyfin again it was like a breath of fresh air after the mess Plex became.

**Comment by hugo-id** [comment]
I just wanted to say this has been _deeply_ overthought (ironically extremely Australian)

PG **must** have a lower age rating than M

PG is in-line with all other PGs around the world, which based on independent websites like commonsensemedia, is usually around the age of 10.

I know it's scary when things aren't spelled out in crystal-clear language, but whoever thought that giving PG an age restriction of 16 is one special little unit

All you have to do to make to make **all these complaints go away** is set PG to an integer **above G** and **below M**

Kind of exactly how it's shown on the government's website?

How peculiar, ey?

**Comment by Shadowghost** [comment]
PG ratings around the world are NOT the same, we have everything from 8 to 15 depending on country.

And as I've explained like 20 times not, just putting it onto something below 15 does not fix the issue. It will only fix the issue if all of your ratings are AU based (which is quite unlikely) because otherwise the levels of another rating system are used which would then not be consistent with the AU ratings.
This issue is only simple if you just look at the AU rating system, but not if you need to guarantee compatibility and coherence between all rating systems. We already have a PR fixing this in master but it is blocked (and needs a rework) once ther EF core PR is merged.

**Comment by ChrisYounger** [comment]
I wrote a script that adds a 'tag' to each movie with its classification. Then it's a very simple matter for me to select which classifications my children can see using the current 'exclude tag' parental controls. I would definitely agree with @Rygle's suggestion above.

**Comment by conradp24** [comment]
It's been a while since I first wrote this, it was more to highlight the
issue, not to complain.
I use Jellyfin exclusively over Kodi.

The issue over PG ratings is having to manually change the PG ratings or
set it higher. The issue with that is my young daughter seeing material she
shouldn't be watching. So, setting it lower is the safety guard. However,
most cartoons or programs appropriate for her are also being blocked.

I would have thought it would be easier, as a total international
resolution, would be to recognise and tag the ratings, but make the rating
system globally editable as categories. Ie. In admin settings the rating
settings of PG, M, 15, etc are globally editable with a user's mnemonic.
Under their settings you can set their age. The ratings section can be
adjusted as a tick box of age range. If the user age matches the age range
in a category it passes.

Alternatively, provide the ability to globally select a grouping of videos
abd make them selectable as accessible by user. The issue with this is that
it doesn't quite work when you have a household of several kids of.varying
range. It would only work by exception.

**Comment by soultaco83** [comment]
I have had a few people on my branch ask about the AU rating as well. I updated the AU.csv for the rating to change PG to age 13. So far the 5 people that have requested this have stated that this is exactly what they want and have had no further requests on the rating so far.

**Comment by Fenruz** [comment]
Could you link to your branch please? It's been such a sad time since they changed it, with full respect to those who give up their time to work on this project, it doesn't change the fact it absolutely ruined it for Australian families :-(

**Comment by soultaco83** [comment]
@Fenruz I want to note: this is a one to one with official jellyfin docker
It has a latest and dev build. The dev build has prs and updates that I find useful and it also contains the efcore database update. The docker backs up the DB and configs when the docker is updated. 
I do not hold any responsibility for the use of this as it was meant for me and then others began using it. 
So far little complains and ones files have been issues with the dev team. 
https://hub.docker.com/r/soultaco83/jellyfin_with_request

**Comment by Rygle** [comment]
Thank you @soultaco83 for your very helpful input!

The fact that this bug is still open is a blatant disregard for the safety of younger people and a lack of empathy for the Jellyfin community because common sense doesn't seem to fit with a database schema.

I have now gone through my whole library and converted everything to pull .nfo files, then I did a text search and replace for anything that says PG or similar and just put in 14. That was the only way I could stop my 13 year old from seeing *R-18+* rated content.

I have also started putting everything M to 15, because my wife and adult daughter don't want to see anything R rated, which was also thoroughly messed up by this insanity.

**Comment by Shadowghost** [comment]
The issue has been acknowledged, discussed and a fix has been proposed and implemented for the next release (merge still pending due to EF core). If you need something faster, for the love of god compile your own version and change the mapping if it's so important to you (I do understand the issues people are having now, but hotfixing it for your usecase might break it for others).
As I've said often enough in this thread: it's not as simple as it looks, we have 20 rating systems to align against each other.

Also a reminder to all of you who keep complaining in whatever kind of manner:
This is a open source project. We do this in our free time. We don't get paid for it. We don't owe you anything. It's not just me who can send in code fixes, anyone of you can. Shouting at us will not get shit done faster.

**Comment by d03j** [comment]
EDIT:  Please ignore. It looks like the user has access to other AU-MA15+ shows but there are some missing. They have access to all libraries but for some reason some shows appear on My Media > Shows but they are all marked as read and when you click on them there's no season info as if the directories were empty. I am trying to get to the bottom of it but it doesn't look related to ratings. I just assumed it did because I just changed this user's max classification and this show didn't appear for them. Apologies!

Any reason a a show that has a "AU-MA15+" classification is being hidden from a user with a maximum allowed parental rating of "MA/MA 15+/MA15+"?

On a parallel note, as mentioned in a previous comment, for Australia MA and MA15 would ideally be separate.
